components/blitz/src/omero/gateway/facility/TablesFacilityHelper.java

```diff
@@ -467,4 +470,198 @@ int getNRows() {
         return gridColumns;
     }
 
+    /**
+     * Update omero.grid.Data with the provided {@link TableData} data.
+     * Note:
+     * - Size and data types must match!
+     * - Size of Double/Float/Long arrays can't be changed!
+     * 
+     * @param toUpdate
+     *            The omero.grid.Data object to update
+     * @param data
+     *            The new data
+     */
+    static void updateData(Data toUpdate, TableData data) {
```

Comment:
Maybe this method could take an optional argument (rows and columns to update) as well, and potentially this could improve performance in scattered updates of huge tables.

Reply:
Not sure that's necessary. `data` doesn't have to be the whole table, `TableData` has internal offsets which specify which particular section of the whole table the `TableData` object represents. My assumption was that the user loads a particular section, modifies the data and then writes that chunk back to the table.